File: TreasurersApp/Models/TransactionTypeViewModel.cs
using Newtonsoft.Json;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TreasurersApp.Models
{
    public class TransactionTypeViewModel
    {
        [JsonProperty("id")]
        public int TransactionTypeID { get; set; }

        [JsonProperty("transactionCategoryId")]
        public int TransactionCategoryID { get; set; }

        [JsonProperty("contributionCategoryDescription")]
        public string TransactionCategoryDescription { get; set; }

        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("description")]
        public string Description { get; set; }

        [JsonProperty("displayOrder")]
        public int DisplayOrder { get; set; }

        [JsonProperty("active")]
        public bool Active { get; set; }
    }
}

File: TreasurersApp/ClientApp/src/app/models/TransactionTypeViewModel.ts
export class TransactionTypeViewModel {
  id: number = null;
  transactionCategoryId: number = null;
  transactionCategoryDescription: string;
  name: string = null;
  description: string = null;
  displayOrder: number = null;
  active = false;

  constructor(id: number,
    transactionCategoryId: number,
    transactionCategoryDescription: string,
    name: string,
    description: string,
    displayOrder: number,
    active: boolean) {
    this.id = id;
    this.transactionCategoryId = transactionCategoryId;
    this.transactionCategoryDescription = transactionCategoryDescription;
    this.name = name;
    this.description = description;
    this.displayOrder = displayOrder;
    this.active = active;
  }
}
